### ISSUE TITLE
GPU-enabled index training

### ISSUE THREAD

**Issue by changhiskhan** [lancedb/lance#1080]
Currently the index training time is dominated by kmeans training, which can be sped up significantly via hardware acceleration.

Recently we added an API to take externally trained IVF centroids. So we could have an optional feature where the centroids are trained using pytorch or rapids then set the centroids explicitly.

Examples:
- https://pypi.org/project/kmeans-pytorch/
- https://medium.com/dropout-analytics/intro-to-k-means-clustering-with-cuml-b6d617e36456

https://github.com/lancedb/lance/blob/main/python/python/lance/dataset.py#L481

**Comment by JudeSurin** [comment]
Hi @changhiskhan, I'm Jude Surin. I'm a Engineering student from Codepath. I'm interested in taking on this issue with GPU enabled Index Training because I would like to build an understanding using RUST and Python. I'd like to contribute and make an impact addressing this issue. Thank you for reading my message @changhiskhan.

**Comment by Sadik00789** [comment]
Hi @changhiskhan, I'd like to work on this issue. 

I plan to add an optional `accelerator` parameter (supporting `"cuda"` via PyTorch) to `create_index` that runs the IVF k-means clustering iterations on GPU and passes the resulting centroids directly into Lance's external centroid API. 

Could you please assign this to me?
